new-components/reporters/slack/internal/reporter/reporter.go

```diff
@@ -0,0 +1,129 @@
+package reporter
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/go-errors/errors"
+
+	"github.com/smithy-security/pkg/env"
+
+	"github.com/smithy-security/smithy/sdk/component"
+	vf "github.com/smithy-security/smithy/sdk/component/vulnerability-finding"
+)
+
+type Conf struct {
+	SlackWebhook string
+}
+
+type slackLogger struct {
+	Conf *Conf
+}
+
+// NewSlackLogger returns a new slack logger.
+func NewSlackLogger(c *Conf) slackLogger {
+	return slackLogger{
+		Conf: c,
+	}
+}
+
+// NewConf returns a new configuration build from environment lookup.
+func NewConf(envLoader env.Loader) (*Conf, error) {
+	var envOpts = make([]env.ParseOption, 0)
+	if envLoader != nil {
+		envOpts = append(envOpts, env.WithLoader(envLoader))
+	}
+
+	webhook, err := env.GetOrDefault(
+		"SLACK_WEBHOOK",
+		"",
+		append(envOpts, env.WithDefaultOnError(false))...,
+	)
+	if err != nil {
+		return nil, errors.Errorf("could not get env variable for SLACK_WEBHOOK: %w", err)
+	}
+	if webhook == "" {
+		return nil, errors.Errorf("variable 'SLACK_WEBHOOK' needs to be set")
+	}
+
+	return &Conf{
+		SlackWebhook: webhook,
+	}, nil
+}
+
+// Report logs the findings summary in slack.
+func (s slackLogger) Report(
+	ctx context.Context,
+	findings []*vf.VulnerabilityFinding,
+) error {
+	logger := component.
+		LoggerFromContext(ctx).
+		With(slog.Int("num_findings", len(findings)))
+
+	if len(findings) > 0 {
+		scanID := findings[0].Finding.FindingInfo.Uid
+		numResults := s.countFindings(findings)
+		startTime := findings[0].Finding.StartTime
+		asTime := time.Unix(*startTime, 0).UTC()
+		toDatetime := asTime.Format(time.RFC3339)
+		msg := fmt.Sprintf("Smithy scan %s, started at %s, completed with %d findings", scanID, toDatetime, numResults)
+		//  TODO: when duplicate enrichment is available, add: out of which, <newResults> new", "the template to use when posting the results"
+		return s.push(msg, s.Conf.SlackWebhook)
+	}
+	logger.Error("Received 0 scans, this is likely an error, skipping sending empty message")
+
+	return nil
+}
+
+func (s slackLogger) push(b string, webhook string) error {
+	type SlackRequestBody struct {
+		Text string `json:"text"`
+	}
+	var err error
+	body, _ := json.Marshal(SlackRequestBody{Text: b})
+	req, err := http.NewRequest(http.MethodPost, webhook, bytes.NewBuffer(body))
```

Comment:
Let's use `http.NewRequestWithContext` and pass the report CTX in input here :)

So we can acknowledge cancellations and in future tracing